contracts/src/KeyManager.sol

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+contract KeyManager is Initializable, OwnableUpgradeable, UUPSUpgradeable {
+    struct CommitteeMember {
+        bytes pubKey;
+        bytes secureChannelKey;
+        bytes dkgEncKey;
+        string networkAddress;
+    }
+
+    struct Committee {
+        uint64 effectiveTimestamp;
+        uint64 id;
+        CommitteeMember[] committeeMembers;
+    }
+
+    event ScheduledCommittee(uint64 indexed id, uint64 effectiveTimestamp, uint64 membersCount, bytes32 membersHash, address indexed scheduledBy);
+    event SetThresholdEncryptionKey(bytes thresholdEncryptionKey, address indexed setBy);
+    event ChangedManager(address indexed oldManager, address indexed newManager, address indexed changedBy);
+
+    error NotManager(address caller);
+    error InvalidAddress(address);
+    error ThresholdEncryptionKeyAlreadySet();
+    error CommitteeIdDoesNotExist(uint64 committeeId, uint64 committeesLength);
+    error InvalidCommitteeMembers();
+    error InvalidEffectiveTimestamp(uint64 effectiveTimestamp, uint64 lastEffectiveTimestamp);
+    error NoCommitteeScheduled(uint64 lastEffectiveTimestamp);
+    error CommitteeIdOverflow();
+    error NoCommitteees();
+
+    bytes public thresholdEncryptionKey;
+    Committee[] public committees;
+    address public manager;
+    uint256[49] private __gap;
+
+    modifier onlyManager() {
+        _onlyManager();
+        _;
+    }
+
+    function _onlyManager() internal view {
+        if (msg.sender != manager) {
+            revert NotManager(msg.sender);
+        }
+    }
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * @notice This function is used to initialize the contract.
+     * @dev Reverts if the manager is the zero address.
+     * @dev Assumes that the manager is valid.
+     * @dev This must be called once when the contract is first deployed.
+     * @param initialManager The initial manager.
+     */
+    function initialize(address initialManager) external initializer {
+        if (initialManager == address(0)) {
+            revert InvalidAddress(initialManager);
```

Comment:
no need to return the invalid address imo. just `InvalidAddress()`, would you agree?